survey/views.py

```diff
@@ -20,37 +22,95 @@
 
 from home.models import Project
 from survey.services import survey_service
+from .forms import InvitationForm
+from .models import Survey
+import logging
 
 from .forms import InvitationForm
 from .models import Survey, SurveyEvidenceSection, SurveyEvidenceFile
 from .services.survey import InvalidInviteTokenException
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponseForbidden
+from .services.survey import SurveyService
+from survey.dashboards.dashboard import get_survey_dashboard
 
 logger = logging.getLogger(__name__)
 
 
 class SurveyView(LoginRequiredMixin, View):
+    login_url = '/login/'
     """
     Manager's view of a survey to be sent out. The manager is able to
     configure what fields are included in the survey on this page.
     """
 
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.survey_service = SurveyService()
+
     def get(self, request: HttpRequest, pk: int):
         return self.render_survey_page(request, pk)
 
-    def post(self, request: HttpRequest, pk: int):
-        return self.render_survey_page(request, pk, is_post=True)
+    def post(self, request, pk):
+        print("POST method called")
+        return self.render_survey_page(request, pk)
 
-    def render_survey_page(self, request: HttpRequest, pk: int, is_post=False):
-        context = {}
-        survey = survey_service.get_survey(request.user, pk)  # Check that we're allowed to get the survey
-        context["survey"] = survey
-        context["first_evidence_section"] = SurveyEvidenceSection.objects.filter(survey=survey).order_by(
-            'section_id').first()
-        context["invite_link"] = survey.get_invite_link(request)
-        context["can_edit"] = {
-            survey.id: survey_service.can_edit(request.user, survey)
+    def get_survey_metrics(self, responses):
+        section_averages = {}
+        response_list = []
```

Comment:
Minor quibble, but in general it's more readable to use the full data type names
```python
section_averages = dict()
response_list = list()
```
to avoid confusion e.g. is `{}` a `set` or a `dict`?